dashboard/app/helpers/aichat_helper.rb

```diff
@@ -0,0 +1,56 @@
+module AichatHelper
+  SAGEMAKER_CLIENT = Aws::SageMakerRuntime::Client.new
+  SAGEMAKER_MODEL_ENDPOINT = "mistral-7b-inst-v01" # "BioMistral-7B"
+  BASE_PROMPT = "You are a helpful chatbot for children. Limit your responses to a small paragraph."
+  ASSISTANT = 'assistant'
+  USER = 'user'
+
+  # The instruction-tuned version of Mistral accepts formatted instructions where conversation roles
+  # must start with a user prompt and alternate between user and assistant.
+  def self.format_inputs_for_sagemaker_request(aichat_params, stored_messages, new_message)
+    stored_messages.push({role: USER, content: new_message})
+    inputs = BASE_PROMPT + aichat_params[:systemPrompt]
+    inputs += aichat_params[:retrievalContexts].join(" ") if aichat_params[:retrievalContexts]
+    inputs = "<s> " + wrap_as_instructions(inputs)
+    # Filter only messages from user or assistant - ignore model update messages.
+    stored_messages.each do |msg|
+      if msg[:role] == USER
+        inputs += wrap_as_instructions(msg[:content])
+      elsif msg[:role] == ASSISTANT
+        inputs += msg[:content] + "</s>"
+      end
+    end
+    {
+      inputs: inputs,
+      parameters: {
+        temperature: aichat_params[:temperature].to_f,
+        max_new_tokens: 300,
+        top_p: 0.9
+      }
+    }
+  end
+
+  def self.request_sagemaker_chat_completion(input_json)
+    SAGEMAKER_CLIENT.invoke_endpoint(
+      endpoint_name: SAGEMAKER_MODEL_ENDPOINT, # required
+      body: input_json.to_json.to_s, # required
+      # target_model: 'mistral-7b-inst.tar.gz',
```

Comment:
Intentionally commented out?

Reply:
I'll remove for now. I'll ask Sanchit about this when he's back in office.

Reply:
@sanchitmalhotra126 - do you mind sharing why `target_model` is commented out? Thanks!

Reply:
Yep you can remove this! This was being used for some testing with [multi-model endpoints](https://docs.aws.amazon.com/sagemaker/latest/dg/multi-model-endpoints.html), where you could specify a single endpoint, and a specific model within that endpoint. For now, we're just using single model endpoints, so only `endpoint_name` is needed